packages/core/data-transfer/src/engine/index.ts

```diff
@@ -183,6 +184,20 @@ class TransferEngine<
       applyTransforms(globalTransforms);
     }
 
+    if (isNumber(throttle) && throttle > 0) {
+      stream = stream.pipe(
+        new PassThrough({
+          objectMode: true,
+          async transform(data, _encoding, callback) {
+            await new Promise((resolve) => {
```

Comment:
Why bother making a promise?
You could have done `setTimeout(() => callback(null, data), throttle)` directly I believe

Reply:
Oops, too late. Can update that in a few minutes.